src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/Base64Transforms.cs

```diff
@@ -25,57 +28,61 @@ public class ToBase64Transform : ICryptoTransform
+
+            if (written != OutputBlockSize)
+                ThrowHelper.ThrowCryptographicException();
 
-            Buffer.BlockCopy(tempBytes, 0, outputBuffer, outputOffset, tempBytes.Length);
-            return tempBytes.Length;
+            Debug.Assert(status == OperationStatus.NeedMoreData);
+            Debug.Assert(consumed == InputBlockSize);
+
+            return written;
         }
 
         public byte[] TransformFinalBlock(byte[] inputBuffer, int inputOffset, int inputCount)
         {
-            ValidateTransformBlock(inputBuffer, inputOffset, inputCount);
+            // inputCount <= InputBlockSize is allowed
+            ThrowHelper.ValidateTransformBlock(inputBuffer, inputOffset, inputCount);
 
             // Convert.ToBase64CharArray already does padding, so all we have to check is that
             // the inputCount wasn't 0
             if (inputCount == 0)
             {
                 return Array.Empty<byte>();
             }
+            else if (inputCount > InputBlockSize)
+            {
+                ThrowHelper.ThrowArgumentInvalidValue(ThrowHelper.ExceptionArgument.inputCount);
+            }
 
             // Again, for now only a block at a time
-            return ConvertToBase64(inputBuffer, inputOffset, inputCount);
-        }
+            Span<byte> input = inputBuffer.AsSpan(inputOffset, inputCount);
+            byte[] output = new byte[OutputBlockSize];
 
-        private byte[] ConvertToBase64(byte[] inputBuffer, int inputOffset, int inputCount)
-        {
-            char[] temp = new char[4];
-            Convert.ToBase64CharArray(inputBuffer, inputOffset, inputCount, temp, 0);
-            byte[] tempBytes = Encoding.ASCII.GetBytes(temp);
-            if (tempBytes.Length != 4)
-                throw new CryptographicException(SR.Cryptography_SSE_InvalidDataSize);
+            OperationStatus status = Base64.EncodeToUtf8(input, output, out int consumed, out int written, isFinalBlock: true);
 
-            return tempBytes;
-        }
+            if (written != OutputBlockSize)
+                ThrowHelper.ThrowCryptographicException();
 
-        private static void ValidateTransformBlock(byte[] inputBuffer, int inputOffset, int inputCount)
-        {
-            if (inputBuffer == null) throw new ArgumentNullException(nameof(inputBuffer));
-            if (inputOffset < 0) throw new ArgumentOutOfRangeException(nameof(inputOffset), SR.ArgumentOutOfRange_NeedNonNegNum);
-            if (inputCount < 0 || (inputCount > inputBuffer.Length)) throw new ArgumentException(SR.Argument_InvalidValue);
-            if ((inputBuffer.Length - inputCount) < inputOffset) throw new ArgumentException(SR.Argument_InvalidOffLen);
+            Debug.Assert(status == OperationStatus.Done);
+            Debug.Assert(consumed == inputCount);
+
+            return output;
         }
 
+        public void Clear() => Dispose();
```

Comment:
Changing which of Clear and Dispose calls which has no practical value or bearing on the stated purpose; please restore it to how it was.